wbooks/src/redux/bookAction/actions.js

```diff
@@ -27,6 +33,31 @@ const actionCreators = {
   },
   bookFilter(value) {
     return { type: actionNames.BOOK_FILTER, payload: { value } };
+  },
+  getRentStatus() {
+    return async (dispatch, getState) => {
+      const currentBookId = getState().book.bookDetail.id;
+      const currentUserId = getState().user.currentUser.id;
+      const response = await BookServices.getRentStatus(currentBookId);
+      if (response.ok) {
+        dispatch({ type: actionNames.GET_RENT_STATUS_SUCCESFUL, payload: { status: response.data, currentUserId } });
+      } else {
+        dispatch({ type: actionNames.GET_RENT_STATUS_FAILURE });
+      }
+    };
+  },
+  postWishlist() {
+    return async (dispatch, getState) => {
+      const currentUserId = getState().user.currentUser.id;
+      const currentBookId = getState().book.bookDetail.id;
+      const response = await BookServices.postWishlist(currentUserId, currentBookId);
+      dispatch({ type: actionNames.POST_WISHLIST });
+      if (response.ok) {
+        dispatch({ type: actionNames.POST_WISHLIST_SUCCESFUL });
+      } else {
+        dispatch({ type: actionNames.POST_WISHLIST_FAILURE });
```

Comment:
`FAILURE` and `SUCCESSFUL` already do the same. IMHO we should use `POST_WISHLIST_FINISHED` instead